### ISSUE TITLE
[http-errors] instanceof HttpError non-functional

### ISSUE THREAD

**Issue by LinusU** [DefinitelyTyped/DefinitelyTyped#19458]
- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @tkrotoff, @BendingBender 

In TypeScript, `instanceof HttpError` fails since `HttpError` is declared as an interface and not a class. I want to do something similar to this:

```js
app.use((err, req, res, next) => {
  if (err instanceof HttpError) {
    // give user nice error
  } else {
    // send generic 500 error
  }
})
```

The javascript in itself is working, but TypeScript is casting the `err` inside of the first conditional branch as `never`.

I tried correcting this by changing `interface HttpError extends Error` to `class HttpError extends Error`, but that caused a few other error to pop up and I think that I'll need some guidance on how to best fix this.

Thank you!

**Comment by BendingBender** [comment]
@LinusU I'm not sure that I can reproduce your problem. The tests for the typings already cover your case of using the `instanceof` operator. And they pass, just take a look into `types/http-errors/http-errors-tests.ts`. The changes I've introduced recently didn't touch this specific part either.

Where do you import the `HttpError` from? Are you sure it's this module?

**Comment by LinusU** [comment]
🤔 Hmm, I might have missed something then, let me get back to you with a fully reproducible test case...

**Comment by LinusU** [comment]
```sh
npm install http-errors express @types/express @types/http-errors
```

```typescript
import { HttpError } from 'http-errors'
import { Request, Response, NextFunction } from 'express'

import express = require('express')

const app = express()

app.use((err: Error, req: Request, res: Response, next: NextFunction) => {
  if (err instanceof HttpError) {

    // Here `err` is typed to "never"
    err

  }
})
```

going in inside `node_modules/@types/http-errors/index.d.ts` and changing `interface` to `class` fixes the problem (not saying that this is the way to go though, just pointing it out 😄)

**Comment by BendingBender** [comment]
I found the root issue. It's because I've tried to return `never` from the `HttpError` constructor. Because it will actually throw when you try to use it to construct a new `HttpError` instance. This way they mark this class as abstract. But there is really no easy way to type it as an abstract class. I've changed the typings such that your use case is covered now, which has much more value than marking the constructor as abstract. I don't expect that there are many people who actually try to construct new `HttpError`s.
